services/codefactor/codefactor-grade.service.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseSvgScrapingService } = require('..')
+const { letterGrades } = require('./codefactor-helpers')
+
+const schema = Joi.object({
+  message: Joi.allow(...Object.keys(letterGrades)),
+}).required()
+
+module.exports = class CodeFactorGrade extends BaseSvgScrapingService {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get route() {
+    return {
+      base: 'codefactor/grade',
+      pattern: ':vcsType(github|bitbucket)/:user/:repo/:branch*',
```

Comment:
As far as I can tell, CodeFactor only appears to work with GitHub and Bitbucket repos

Reply:
Yep. Thanks!!!